### PR TITLE
Qubit specifier

### LINKED ISSUE #15: Qubit specifier for POVM implementations
So far, when we instantiate a `POVMImplementation` object, we set a number of qubits and assume that the POVM will act on all of the qubits of the circuit that will later be supplied.

Would it be interesting to be able to initialize `POVMImplementation` objects by specifying a subset of qubits to be measured (for instance as a sequence of `QubitSpecifier` or a list of integers) ?

### PR DESCRIPTION
**Pull request by timmintam**
Offer the possibility to the user to specify on which qubits a `POVMImplementation` instance should act (possibly not on all qubits for instance).

Closes #15

### CODE REVIEW

File: povm_toolbox/library/povm_implementation.py [hunk 1]
```diff
@@ -41,15 +41,18 @@ class POVMImplementation(ABC, Generic[MetadataT]):
...
         """Initialize the POVMImplementation.
 
         Args:
             n_qubit: number of logical qubits in the system.
+            qubit_specifier: list of index specifying on which qubits the POVM acts.
         """
         super().__init__()
         self.n_qubit = n_qubit
         # TODO: add qubit_specifier or layout to apply, see issue #15
+        self.idx_layout = qubit_specifier
```
Comment (mrossinek) [inline povm_toolbox/library/povm_implementation.py:55]: If we make this a public attribute, we should find a good name. As of now, I find it confusing to see a name different than the `__init__` argument.
That said, not sure if `qubit_specifier` is ideal. I am open to suggestions.

Maybe even `layout` or `measurement_layout`?

File: povm_toolbox/library/povm_implementation.py [hunk 2]
```diff
@@ -110,15 +113,18 @@ def compose_circuits(self, circuit: QuantumCircuit) -> QuantumCircuit:
         dest_circuit = circuit.copy()
         dest_circuit.remove_final_measurements(inplace=True)
 
-        if dest_circuit.layout is None:
-            # Basic one-to-one layout
-            index_layout = list(range(dest_circuit.num_qubits))
+        if self.idx_layout is None:
+            if dest_circuit.layout is None:
+                # Basic one-to-one layout
+                index_layout = list(range(dest_circuit.num_qubits))
```
Comment (mrossinek) [inline povm_toolbox/library/povm_implementation.py:119]: This default should be explicitly documented in the docstring.

File: povm_toolbox/library/povm_implementation.py [hunk 3]
```diff
@@ -110,15 +113,18 @@ def compose_circuits(self, circuit: QuantumCircuit) -> QuantumCircuit:
...
 
-        elif isinstance(dest_circuit.layout, TranspileLayout):
-            # Extract the final layout of the transpiled circuit (ancillas are filtered).
-            index_layout = dest_circuit.layout.final_index_layout(filter_ancillas=True)
+            elif isinstance(dest_circuit.layout, TranspileLayout):
+                # Extract the final layout of the transpiled circuit (ancillas are filtered).
+                index_layout = dest_circuit.layout.final_index_layout(filter_ancillas=True)
```
Comment (mrossinek) [inline povm_toolbox/library/povm_implementation.py:123]: Also this "fallback" should be mentioned.

**Review by mrossinek — COMMENTED** [review]
Overall this looks good. Some suggestions on the how-to guide, otherwise some naming.

File: povm_toolbox/library/povm_implementation.py (hunk 1 above)
Comment (timmintam) [reply povm_toolbox/library/povm_implementation.py:55]: `measurement_layout` sounds good 👍

**Review by mrossinek — APPROVED** [review]
LGTM :+1:

**Pull request merged**